src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -203,5 +203,27 @@ public void ProcessValidationErrors(IEnumerable<IdentityError> errors, User user
                 }
             }
         }
+
+        #region UserClaimsPrincipalFactory<IUser>
+
+        protected override async Task<ClaimsIdentity> GenerateClaimsAsync(IUser user)
```

Comment:
More questions on this one, and summoning @PinpointTownes  too

- Could we have it in a separate class instead, dedicated to this interface?
- Should this implementation be part of the facebook module, or the users module? Meaning are these claims for facebook only, or is it something more general?
- Assuming this is not only for facebook, do we have to use these claims identifier or should/can we use OpenId ones like here: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/UserInfoController.cs#L65
- When we resolve this interface (we already do in MembershipService) should we resolve an enumeration?
- I don't see any usage of this interface, what component is using it right now, and how is it relevant for Facebook?

I think these questions will trigger more refactoring, at least for `IMembershipService` which I think is using it incorrectly, and for the other services which should actually use it instead of `IMembershipService`.